pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4668,4 +4669,38 @@ private void internalGetReplicatedSubscriptionStatusForNonPartitionedTopic(Async
             resumeAsyncResponseExceptionally(asyncResponse, e);
         }
     }
+
+    protected CompletableFuture<SchemaCompatibilityStrategy> internalGetSchemaCompatibilityStrategy() {
+        validateTopicOperation(topicName, TopicOperation.GET_SCHEMA_COMPATIBILITY_STRATEGY);
+
+        return getTopicPoliciesAsyncWithRetry(topicName)
+                .thenApply(op -> op.map(TopicPolicies::getSchemaCompatibilityStrategy)
```

Comment:
This logic seems not right.
If we have topic level policy setting as `SchemaCompatibilityStrategy.UNDEFINED`, we should use namespace level setting.

Reply:
And broker level setting is missing here.